commands.go

```diff
@@ -79,6 +79,11 @@ func Commands(metaPtr *command.Meta) map[string]cli.CommandFactory {
 				Meta: meta,
 			}, nil
 		},
+		"keygen": func() (cli.Command, error) {
```

Comment:
What internal interface do you want to have to perform the queries? In Consul there is an [rpc_client](https://github.com/hashicorp/consul/blob/master/command/keyring.go#L60), but it is also reused in other commands. I thought of adding an http endpoint, like it is done for other commands, but wanted to know your opinion first.